docs/canonical-ways-to-investigate-bugs.md

```diff
@@ -0,0 +1,368 @@
+```
+
+### A problem during "Add Account"
+
+The _Add Account_ flow involves the [`currencyBridge.scanAccounts`](CurrencyBridge.md) logic which itself is mostly about performing synchronisation. The main problem is that you might not know the XPUB at this point in time and it might breaks before. What we can do easily however is to reuse the APDUs that appears in the log and replay them against the CLI.
+
+On top of the /logsviewer page, there is a handy button "Export APDUs" that we can use to retrieve ALL the apdus. Now, you need to locate the part that concerns the add accounts.
+
+For instance, in this example we are investigating a ZEN scan accounts and here is the binary exchange with the device of that part: (from my testing accounts)
+
+```
+=> e040000009028000002c80000079
+<= 410486f3033309829d1f1458622c6d08430f7130d4d1b2ddef378dfe4c1dd025dbbbeb51298dee5e7ad0c31a1f555fe8a2941bb0018185697276861622d0a3b1bcad237a6e6b346975357a65554a39616873554b677735635470446f374b32656554417a3473271f6a912e786b63d98809c92ba5c58beac445c698dcadb8c8e91652ce64745f9000
+=> e016000000
+<= 208920960407486f72697a656e035a454e9000
+=> e040000009028000002c80000079
+<= 410486f3033309829d1f1458622c6d08430f7130d4d1b2ddef378dfe4c1dd025dbbbeb51298dee5e7ad0c31a1f555fe8a2941bb0018185697276861622d0a3b1bcad237a6e6b346975357a65554a39616873554b677735635470446f374b32656554417a3473271f6a912e786b63d98809c92ba5c58beac445c698dcadb8c8e91652ce64745f9000
+=> e04000000d038000002c8000007980000000
+<= 41046a05e02c73991c84f42444cfa4b3bc1046838fe609a220b1498d766e23264d57ea4c4acaae65398c13adad5d045324e03abdeec9b3fef1d88d4abb832a44af9a237a6e53467933565a58506a48366b5944736b78544e65677a67365675634258317738466e39f205deb0a0a9763b86fecc61046a453be8a6763c8bdbceb8b3d8abff46459000
+=> e04000000d038000002c8000007980000001
+<= 4104392c448b6a725c7c8c32993d0483e38927830dfec299b74963d264b10e9db900215b1bf058a992604c86124f5d40c178b805da0f8af0889814cb7ffb011b4308237a6e62476d615656456d676d317a416e7a74724172617370506177574a73506a527a4af6c3ce6238696dc78ee8d57b161b25d79b89ace20515966789ac09f79c713cbd9000
+```
+
+We can save that to a `apdus.txt` file.
+
+Now, we are going to our Terminal and we are going to replay that exact series of device exchanges. As everything is deterministic here, it should just works™.
+
+We do this:
+
+```
+ledger-live proxy -f apdus.txt
+```
+
+which starts a server that will be able to respond like a device. It should logs something like
+
+```
+5 mocked APDUs will be replayed from /Users/grenaudeau/Desktop/logs.txt
+DEVICE_PROXY_URL=ws://localhost:8435
+DEVICE_PROXY_URL=ws://192.168.0.25:8435
+
+Nano S proxy started on 192.168.0.25
+```
+
+Now, we go to a new Terminal and we do the actual scan accounts via the `ledger-live sync -c ZEN` command. But we also need to make sure to use the environment DEVICE_PROXY_URL to use the proxy instead of a real device. (btw make sure to have removed `dbdata` folder that is used by ledger-live CLI)
```

Comment:
```suggestion
Now, we go to a new Terminal and we do the actual scan accounts via the `ledger-live sync -c ZEN` command. But we also need to make sure to use the environment `DEVICE_PROXY_URL` to use the proxy instead of a real device. (btw make sure to have removed `dbdata` folder that is used by ledger-live CLI)
```